src/elements/e-video.vue

```diff
@@ -0,0 +1,254 @@
+    name: 'e-video',
+    status: 0, // TODO: remove when component was prepared for current project.
+
+    // components: {},
+
+    props: {
+      /**
+       * Allows the user to watch the video in fullscreen
+       */
+      allowfullscreen: {
+        type: Boolean,
+        default: false,
+      },
+
+      /**
+       * Specifies the source of the video
+       *
+       * We can decide between youtube and vimeo
+       */
+      source: {
+        type: String as PropType<'youtube' | 'vimeo'>,
+        required: true,
+        validator(value: string) {
+          return Object.keys(playerUrlSources).includes(value);
+        },
+      },
+
+      /**
+       * Allows to add/remoe the border around the player
+       *
+       * When set to 0 the border is completely removed
+       *
+       * Default from iframe is 2
+       */
+      frameborder: {
+        type: [String, Number],
+        default: 2,
+      },
+
+      /**
+       * Accepts a valid url from a youtube or vimeo video
+       */
+      videoUrl: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Accepts an ID from a youtube or vimeo video
+       */
+      videoId: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Allows to set the video player width.
+       */
+      width: {
+        type: [String, Number],
+        default: 560,
+      },
+
+      /**
+       * Allows to set the video player height.
+       */
+      height: {
+        type: [String, Number],
+        default: 315,
+      },
+
+      /**
+       * Defines the players size to be responsive 100% width to the parent container
+       */
+      responsive: {
+        type: Boolean,
+        default: false,
+      },
+    },
+
+    // emits: {},
+    // setup(): Setup {},
+
+    computed: {
+      mapSrc(): string | undefined {
+        const playerBaseUrl = playerUrlSources[this.source];
+
+        if (!(this.videoId || this.videoUrl)) {
+          // eslint-disable-next-line no-console
+          console.error("Neither 'videoId' or 'videoUrl' were defined", this.$el);
+
+          return undefined;
+        }
+
+        if (this.videoUrl && !this.videoId) {
+          if (!this.validateUrl()) {
+            return undefined;
+          }
+
+          if (this.source === 'youtube') {
+            /**
+             * Extract id from video url
+             */
+            const id = this.videoUrl.substring(
+              this.videoUrl.indexOf('v=') + 2,
+              this.videoUrl.indexOf('&')
+            );
+
+            return `${playerBaseUrl}${id}`;
+          }
+
+          const id = this.videoUrl.substring(
+            this.videoUrl.indexOf(videoUrlSources[this.source])
+              + videoUrlSources[this.source].length,
+            this.videoUrl.length
+          );
+
+          return `${playerBaseUrl}${id}`;
+        }
+
+        if (this.videoId.includes(videoUrlSources[this.source])) {
```

Comment:
todo: I think this test is insufficient. It would be possible to inject anything into the url as long as it does not match the video domains. I would recommend to use a regext test for `[0-9a-Z_]` instead.

Reply:
Check for all possible characters in vimeo and youtube ids

Reply:
According to https://webapps.stackexchange.com/questions/54443/format-for-id-of-youtube-video the videoId is a 11 character string matching `[0-9A-Za-z_-]{10}[048AEIMQUYcgkosw]`